test-framework/sudo-compliance-tests/src/su/flag_shell.rs

```diff
@@ -50,14 +49,17 @@ fn specified_shell_does_not_exist() -> Result<()> {
     assert!(!output.status().success());
     assert_eq!(Some(127), output.status().code());
 
-    let diagnostic = "su: failed to execute /does/not/exist: No such file or directory";
+    let diagnostic = if sudo_test::is_original_sudo() {
+        format!("su: failed to execute /does/not/exist: No such file or directory")
```

Comment:
clippy complains about these `format!` calls. either you could store the string "/does/not/exist" in a variable and use string interpolation here (and also use the variable in the `Command` call above); or you could switch these to string literals